File: backend/app/schemas/workbook.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.worksheet import WorksheetRead


class WorkbookRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    file_size_bytes: int | None
    created_at: datetime
    updated_at: datetime


class WorkbookDetail(WorkbookRead):
    worksheets: list[WorksheetRead]


class WorkbookUpdate(BaseModel):
    filename: str

File: src/types/workbook.ts
import type { WorksheetRead } from "./worksheet";

export interface WorkbookRead {
  id: string;
  filename: string;
  file_size_bytes: number | null;
  created_at: string;
  updated_at: string;
}

export interface WorkbookDetail extends WorkbookRead {
  worksheets: WorksheetRead[];
}
